datafusion/datasource/src/file_stream.rs

```diff
@@ -91,7 +91,7 @@ impl FileStream {
         let files = config.file_groups[partition].clone();
 
         Ok(Self {
-            file_iter: files.into(),
+            file_iter: files.iter().cloned().collect(),
```

Comment:
let file_group = config.file_groups[partition].clone();

use `file_group.files.into_iter().collect()` to avoid clone